### DISCUSSION TITLE
Should we create separate files for services like Plex, Local Now, etc.?

### DISCUSSION THREAD

**Discussion by freearhey** [iptv-org/community#520] [Ideas]
What you think?

1. Good idea 👍
2. Bad idea 👎
3. Don't know 😕

**Comment by yarafie** [comment]
1

**Comment by admin404** [comment]
best idea

**Comment by fotvnetwork** [comment]
👍

**Comment by themak1985** [comment]
3
Not sure what it would look like or why we would do it - Plex, Samsung, Pluto, Local Now, etc. 
I also think it would make takedown requests much more prevalent.

**Reply by freearhey** [reply]
I was thinking about creating separate files for each service in the `channels/` folder and moving the channels broadcast by these services to them. It should look something like this:

**channels/stirr.m3u**

```
#EXTM3U
#EXTINF:-1 tvg-id="Stadium.stirr" tvg-country="US" tvg-language="English" tvg-logo="https://i.imgur.com/n38w3FD.png" group-title="Sports",Stadium (720p)
https://dai.google.com/linear/hls/event/0jRU1DBXSW6a_TFheLfAUQ/master.m3u8
```

**Reply by themak1985** [reply]
So it wouldn't create new playlists? Just for easier organizing/work? If so, then yes.

**Reply by freearhey** [reply]
Yes, that's right.

**Comment by Sphinxroot** [comment]
good idea   personnaly i would separate them by country as well  ex  br_pluto.m3u  es_pluto.m3u  and so on  for every services  ,  yes it will create a lot of new list  ,but  loading them and updating them will be easier

**Reply by freearhey** [reply]
maybe we can start by separating into services for now, and if that's not enough, then we'll split these files into countries as well

**Comment by Dum4G** [comment]
If us playlist will become editable in web again I'm into it.

**Reply by freearhey** [reply]
As I understand it, for this it should be enough for the file to be less than 1Mb in size.

https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/limits-for-viewing-content-and-diffs-in-a-repository#text-limits

**Reply by BellezaEmporium** [reply]
More than enough to be quite honest.

US channels are quite bloated by the Pluto / STIRR / XUMO channels.

**Comment by what-a** [comment]
I'm thinking PBS is another contender for it's own file (separate from us.m3u). We currently have 118 PBS channels. I'm planning on adding another 23+ which will put us at 140+ PBS channels. It certainly would be easier to scrub and maintain if they were in their own file.

**Reply by freearhey** [reply]
Just to clarify, at the moment channels are divided only by broadcast source, not by name. So we can collect all channels broadcast through pbs.org (100+ channels), but it will not be all PBS channels. But if it helps, I can create such a file.

**Comment by what-a** [comment]
Those that I'm planning on adding are all through pbs.org, so I think we'll end up with about 130 total via pbs.org.   That seems sufficient enough to justify a separate file. Looking forward to it. Thanks.

**Reply by freearhey** [reply]
Done. 

https://github.com/iptv-org/iptv/blob/master/channels/us_pbs.m3u

**Reply by what-a** [reply]
Thank you!    BTW, how do we handle alternate working streams for the same channel and even in the same quality (i.e. only difference being two different broadcast sources)?.    I've only come across this twice, but am wondering how best to handle it. Should I input two separate entries (same as if they were two separate channels), or stack the steam links one line directly after the other, or?

**Reply by Dum4G** [reply]
The first option is valid

**Reply by what-a** [reply]
Got it!  Thanks!

**Comment by allanf181** [comment]
keeping on separate lists would help me with that: https://github.com/iptv-org/iptv/discussions/4783

**Comment by loucasal** [comment]
As a user, is there a way of retaining the same playlist structure that the repo has now adopted? Using the main m3u link still groups channels by category for me.

**Reply by Dum4G** [reply]
Don't use repo itself. We have autogenerated sorted lists available in the description

**Reply by loucasal** [reply]
Maybe I wasn't clear. https://iptv-org.github.io/iptv/index.m3u groups channels by category. What should I use to keep the same structure as in the repo (countries + services)?

**Reply by Dum4G** [reply]
So far we don't have an index file that separates services from countries. @freearhey may consider making one

**Reply by freearhey** [reply]
If the rest of the community will support the idea, we can start grouping channels in the main playlist this way.
